### PR TITLE
Add chefspec tests for default recipe

### PR DESCRIPTION
**Pull request by lo9key**
Updates cookbook using `chef generate`, which created kitchen.yml and metadata files, among others.
Adds chefspec tests for the default recipe.

One thing I'm not sure about is which platforms this cookbook should run on.  `chef generate` assumed centos and debian, but maybe that's not true?

@osuosl-cookbooks/chefs  please review.

```
$ rake
Running RuboCop...
Inspecting 6 files
......

6 files inspected, no offenses detected
chef exec foodcritic --epic-fail any .

chef exec rspec

aliases::default
  centos 6.7
    should add magic_shell_alias "ll"
    should add magic_shell_environment "EDITOR"
  centos 7.2.1511
    should add magic_shell_alias "ll"
    should add magic_shell_environment "EDITOR"
  debian 8.4
    should add magic_shell_alias "ll"
    should add magic_shell_environment "EDITOR"

Finished in 0.21807 seconds (files took 1.55 seconds to load)
6 examples, 0 failures


ChefSpec Coverage report generated...

  Total Resources:   2
  Touched Resources: 2
  Touch Coverage:    100.0%

You are awesome and so is your test coverage! Have a fantastic day!
```

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 name             'aliases'
+issues_url       'https://github.com/osuosl-cookbooks/aliases/issues'
+source_url       'https://github.com/osuosl-cookbooks/aliases'
 maintainer       'Oregon State University'
 maintainer_email 'rudy@grigar.net'
```
Comment (ramereth) [inline metadata.rb]: This needs to be updated to ``chef@osuosl.org``

File: test/integration/default/serverspec/server_spec.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+require 'serverspec'
+
+set :backend, :exec
```
Comment (ramereth) [inline test/integration/default/serverspec/server_spec.rb]: Can you please go ahead and add the relevant serverspec tests we need here? No time like the present to learn!

File: metadata.rb [hunk 3]
```diff
@@ -2,9 +2,11 @@
...
 maintainer       'Oregon State University'
-maintainer_email 'rudy@grigar.net'
+maintainer_email 'chef@osuosl.org'
 license          'Apache 2.0'
 description      'Installs/Configures aliases'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
 version          '0.1.2'
 depends          'magic_shell', '>= 1.0.0'
+supports         'centos'
+supports         'debian'
```
Comment (ramereth) [inline metadata.rb]: You need to add what version of the OS it supports. For CentOS you need two lines for each major version.

**Comment by lo9key** [comment]
@ramereth done.  Is this something I should be watching out for in future serverspec tests?  I can see that `apt` appears in the error, which I guess was my indication that it's required and should be a dependency.

```
       [2016-11-09T16:47:54+00:00] ERROR: Cookbook apt not found. If you're loading apt from another cookbook, make sure you configure the dependency in your metadata
       [2016-11-09T16:47:54+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
>>>>>> Converge failed on instance <default-debian-8>.
>>>>>> Please see .kitchen/logs/default-debian-8.log for more details
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: SSH exited (1) for command: [sh -c '

sudo -E /opt/chef/bin/chef-solo --config /tmp/kitchen/solo.rb --log_level auto --force-formatter --no-color --json-attributes /tmp/kitchen/dna.json
']
>>>>>> ----------------------
zlib(finalizer): the stream was freed prematurely.
```

**Comment by ramereth** [comment]
@lo9key yes, but it should rarely happen.

**Pull request merged**